controllers/spacebindingcleanup/spacebinding_cleanup_controller_test.go

```diff
@@ -34,13 +35,88 @@ func TestDeleteSpaceBinding(t *testing.T) {
 	sbLaraRedhatAdmin := sb.NewSpaceBinding("lara", "redhat", "admin", "signupA")
 	sbJoeRedhatView := sb.NewSpaceBinding("joe", "redhat", "view", "signupB")
 	sbLaraIbmEdit := sb.NewSpaceBinding("lara", "ibm", "edit", "signupC")
+	sbPublicViewerRedhatView := sb.NewSpaceBinding("public-viewer", "redhat", "view", "signupD")
 
 	redhatSpace := spacetest.NewSpace(test.HostOperatorNs, "redhat")
 	ibmSpace := spacetest.NewSpace(test.HostOperatorNs, "ibm")
 
 	laraMur := masteruserrecord.NewMasterUserRecord(t, "lara")
 	joeMur := masteruserrecord.NewMasterUserRecord(t, "joe")
 
+	t.Run("publicViewer config is enabled", func(t *testing.T) {
```

Comment:
minor - but should we create a test for 
```
Config: toolchainv1alpha1.PublicViewerConfig{
				Enabled:  true,
				Username: "",
			},
```
when the Username is empty ? WDYT ?

Reply:
good catch. What would be here the expected behavior? panic?
To prevent this from happening I'd propose to have some default and validation, like follows

```diff
// PublicViewer used to enable public-viewer support
// +k8s:openapi-gen=true
type PublicViewerConfig struct {
	// +kubebuilder:default:=false
	Enabled bool `json:"enabled"`

	// +optional
+	// +kubebuilder:validation:MinLength=4
	// +kubebuilder:validation:MaxLength=63
	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	// +kubebuilder:default:=public-viewer
	Username string `json:"username,omitempty"`
}
```

Reply:
WDYT?

Reply:
TBH maybe we could remove the `Enabled` field and just use the `username`, if provided then the feature is enabled. This will reduce the amount of permutations to check/test. Would also simplify a bit the configuration.

Reply:
The default validation using kubebuilder markers works fine, but also just checking that the `len(username) > 0` , in case it's empty we can assume that the feature is disabled.

Reply:
I'd like to have validation where possible, but if we want to deduce enablement from the `Username` field only, then we're going to have to change these validation rules.

Maybe change the validation pattern to this?
```go
// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?)?$`
```

This will make the empty string valid, but still validate that usernames must be shaped like a username if they're set.